frontend/src/app/settings/modelsources/t4c-settings/service/t4c-repos/t4c-repo.service.ts

```diff
@@ -5,76 +5,133 @@
-      .get<T4CServerRepository[]>(this.urlFactory(instance_id))
-      .pipe(tap((repositories) => this._repositories.next(repositories)));
+  loadRepositories(instance_id: number): void {
+    this.http
+      .get<T4CServerRepository[]>(this.backendURLFactory(instance_id))
+      .subscribe({
+        next: (repositories) => this._repositories.next(repositories),
+        error: () => this._repositories.next(undefined),
+      });
   }
 
-  createT4CRepository(
-    instance_id: number,
-    repository: CreateT4CRepository
-  ): Observable<T4CRepository> {
-    return this.http.post<T4CRepository>(
-      this.urlFactory(instance_id),
-      repository
-    );
+  refreshRepositories(instance_id: number): void {
+    const updateRepositories = this._repositories.value?.map((repository) => {
+      return { ...repository, status: 'LOADING' } as T4CServerRepository;
+    });
+    this._repositories.next(updateRepositories);
+    this.loadRepositories(instance_id);
   }
 
-  deleteRepository(
+  createRepository(
     instance_id: number,
-    repository_id: number
-  ): Observable<null> {
-    return this.http.delete<null>(
-      `${this.urlFactory(instance_id)}${repository_id}/`
-    );
+    repository: CreateT4CRepository
+  ): Observable<T4CRepository> {
+    return this.http
+      .post<T4CRepository>(this.backendURLFactory(instance_id), repository)
+      .pipe(tap(() => this.loadRepositories(instance_id)));
   }
 
   startRepository(
     instance_id: number,
     repository_id: number
   ): Observable<null> {
-    return this.http.post<null>(
-      `${this.urlFactory(instance_id)}${repository_id}/start/`,
-      {}
-    );
+    this.publishRepositoriesWithChangedStatus(repository_id, 'LOADING');
+    return this.http
+      .post<null>(
+        `${this.backendURLFactory(instance_id)}${repository_id}/start/`,
+        {}
+      )
+      .pipe(tap(() => this.loadRepositories(instance_id)));
   }
 
   stopRepository(instance_id: number, repository_id: number): Observable<null> {
-    return this.http.post<null>(
-      `${this.urlFactory(instance_id)}${repository_id}/stop/`,
-      {}
-    );
+    this.publishRepositoriesWithChangedStatus(repository_id, 'LOADING');
+    return this.http
+      .post<null>(
+        `${this.backendURLFactory(instance_id)}${repository_id}/stop/`,
+        {}
+      )
+      .pipe(tap(() => this.loadRepositories(instance_id)));
   }
 
   recreateRepository(
     instance_id: number,
     repository_id: number
   ): Observable<null> {
-    return this.http.post<null>(
-      `${this.urlFactory(instance_id)}${repository_id}/recreate/`,
-      {}
-    );
+    this.publishRepositoriesWithChangedStatus(repository_id, 'LOADING');
+    return this.http
+      .post<null>(
+        `${this.backendURLFactory(instance_id)}${repository_id}/recreate/`,
+        {}
+      )
+      .pipe(tap(() => this.loadRepositories(instance_id)));
+  }
+
+  deleteRepository(
+    instance_id: number,
+    repository_id: number
```

Comment:
```suggestion
    instanceId: number,
    repositoryId: number
```
Please update all parameters in this service to camelCase.

Reply:
I already did this while handling the previous comment since I also noticed it there 😄